lib/service/service.go

```diff
@@ -1447,8 +1447,16 @@ func (process *TeleportProcess) initAuthService() error {
 			warnOnErr(tlsServer.Close(), log)
 		} else {
 			log.Info("Shutting down gracefully.")
-			ctx := payloadContext(payload, log)
-			warnOnErr(tlsServer.Shutdown(ctx), log)
+			// NOTE: Graceful shutdown of auth.TLSServer is disabled right now, because we don't
+			// have a good model for performing it.  In particular, watchers and other GRPC streams
+			// are a problem.  Even if we distinguish between user-created and server-created streams
+			// (as is done with ssh connections), we don't have a way to distinguish "service accounts"
```

Comment:
We could encode this info into client certificates? E.g. those produced by tsh login can be considered "user" and identity files produced by tctl sign can be considered "robot". Not in this PR, in future.

Reply:
+1

Reply:
Definitely.  In fact, `CertExtensionImpersonator` _almost_ does this already.